### DISCUSSION TITLE
Card

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#48] [Feedback]
Use this place to discuss the [card component](https://ons-design-system.netlify.com/components/card) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by dudelmeister** [comment]
This is just a suggestion - Currently the cards are a container of a link with a description - I can't help but feel as if the click area should be the entire card itself... How much research have we done on this component? I'm assuming it's tested ok?

**Comment by jrbarnes9** [comment]
Cards broken on iOS 10 and under:
![Screenshot 2019-04-24 at 12 13 37](https://user-images.githubusercontent.com/43346934/56655657-5095e980-668b-11e9-97bc-be1238ee922e.png)

**Comment by jrbarnes9** [comment]
The h3 font size on cards is too large for long titles 
<img width="384" alt="Screenshot 2019-05-13 at 14 21 54" src="https://user-images.githubusercontent.com/43346934/57624752-7f1c2b80-758a-11e9-9f5e-0c4cc940db69.png">

**Comment by trevorsaint** [comment]
The card component needs the flexibility for changing heading level.

**Comment by jrbarnes9** [comment]
`TextId` missing from Card component macro options table

**Comment by jrbarnes9** [comment]
I suspect the `aria-labelledBy` and `aria-describedBy` attributes are not valid/required on the `div` because it's not an interactive element. Perhaps a hangover from the original implementation which was changed after DAC audit #1321 ?

**Reply by jrbarnes9** [reply]
To be reviewed by DAC in May2022 audit.

**Comment by Al-Davies** [comment]
**Card - UR Findings:**

- Descriptive text needs to be clear on what is contained within the links
- If it missing vital information, then users may ignore the link as the link title and description don't appear to be what they need.
- Headings need to be clear i.e. Cause of death is clear, but Population characteristics is vague.
- Help page on Census had a card titled 'How to complete the census' it held much more than this, like how to help others, when to complete it, moving house etc.
- Users scan pages looking for words, so any help text needs to be as wide-covering as possible to help explain the content that is available

**Comment by AtinDESNEZ** [comment]
I was wondering if card background (Card set with Headline figures) i.e. grey color has been approved for accessibility when grey being foreground on default white color pages. I was planning to use it for our service and is not sure from accessibility perspective.

**Reply by MagdalenaLarge** [reply]
Hello @AtinDESNEZ thank you for contacting us.
We recently conducted an accessibility review that included this specific component. While the light grey on white does have low contrast, it is considered acceptable for decorative grouping purposes—as long as it’s not used to convey information independently. All other content within that box meets accessibility standards. I hope this clarifies things. Please don’t hesitate to reach out if you have any further questions.
Thanks.

**Reply by AtinDESNEZ** [reply]
Thanks for the quick response. Appreciate it!

**Reply by AtinDESNEZ** [reply]
What's the correct command (npm install ons-design-system) to import ONS design system components.

**Reply by rmccar** [reply]
All the documentation to get started using the DS with npm should be [here](https://service-manual.ons.gov.uk/design-system/guidance/install-with-npm)

**Comment by AtinDESNEZ** [comment]
[like]  Gupta, Atin (Energy Security) reacted to your message:
________________________________

**Comment by AtinDESNEZ** [comment]
Have you had a chance to test the card in a disabled form like button so that user knows what's coming up or will get activated once they have done the requested task.

**Reply by MagdalenaLarge** [reply]
Hello @AtinDESNEZ ,
Card component was designed to display specific data in a dashboard format. You can read more about how this component is intended to be used within ONS: https://service-manual.ons.gov.uk/content/content-types/statistical-articles#headline-facts-and-figures-optional-
We didn’t intend for it to have any functionality beyond what it currently offers.

@rmccar - are you able to advise if the behaviour explained above is possible using card component?

**Reply by AtinDESNEZ** [reply]
Appreciate the clarity. Thanks. 

@rmccar Please do let me know if its possible to implement the disabled feature (as happens in button) on these cards. THanks.

**Reply by rmccar** [reply]
Hi, the card component wasn't designed with it being used in disabled forms in mind. Usually it would be an input/button element that in some cases you would want to disable in a form. The card component just includes some information and a link. Is it the link you would want to disable? I'd need more details on your use case

**Reply by AtinDESNEZ** [reply]
Thanks. I hav 2-3 sections which I want user to edit & update in a sequential way. They are dependent on completion stage of each other, so wanted to show the cards in a disabled state till they complete prev one. That way user would always know what;s coming up.

**Reply by rmccar** [reply]
I think I might need you to take me through your use case. From that it sounds like you would want to use a new page per section to me as that is a pattern we have used in the past with questionnaires in eQ. Then you would use the summary component to show them how many sections they had completed and redirect to the summary page after each section. The card component isn't really designed for this use. But happy to have a chat
